File: Algorithm/Baekjoon/Python/SilverII/2885.py
k = int(input())

n=1
while(True):
  n*=2 
  if k<=n:
    break
print(n, end=' ')

d=0
cnt = 0
while(k>0):
  if k>=n:
    k-=n
  else:
    d += n//2
    n //=2
    cnt+=1
print(cnt)

File: Algorithm/Baekjoon/Java/Greedy/2885.java
import java.util.Scanner;

class Main {
  public static void main(String[] args) {
        Scanner scan = new Scanner(System.in); 
        int k = scan.nextInt();

        int n = 1;
        while(true){
          n *= 2;
          if(k<=n) break;
        }
        System.out.print(n+" ");
        int d = 0;
        int cnt = 0;
        while(k>0){
          if(k>=n){
            k-=n;
          }else{
            n /= 2;
            cnt++;
          }
        }
        System.out.println(cnt);
  }
}
